tests/integration/security/jwt_test.go

```diff
@@ -465,3 +468,182 @@ func TestRequestAuthentication(t *testing.T) {
 			}
 		})
 }
+
+// TestRequestAuthentication tests beta authn policy for jwt on ingress.
+// The policy is also set at global namespace, with authorization on ingressgateway.
+func TestRequestAuthenticationOnIngress(t *testing.T) {
+	framework.NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			var ingr ingress.Instance
+			var err error
+			if ingr, err = ingress.New(ctx, ingress.Config{
+				Istio: ist,
+			}); err != nil {
+				t.Fatal(err)
+			}
+			t.Log(ingr)
+
+			ns := namespace.NewOrFail(t, ctx, namespace.Config{
+				Prefix: "req-authn",
+				Inject: true,
+			})
+
+			// Apply the policy.
+			namespaceTmpl := map[string]string{
+				"Namespace":     ns.Name(),
+				"RootNamespace": rootNamespace,
+			}
+
+			applyPolicy := func(filename string, ns namespace.Instance) []string {
+				policy := tmpl.EvaluateAllOrFail(t, namespaceTmpl, file.AsStringOrFail(t, filename))
+				g.ApplyConfigOrFail(t, ns, policy...)
+				return policy
+			}
+
+			securityPolicies := applyPolicy("testdata/requestauthn/global-jwt.yaml.tmpl", rootNS{})
+			ingressCfgs := applyPolicy("testdata/requestauthn/ingress.yaml.tmpl", ns)
+
+			defer g.DeleteConfigOrFail(t, rootNS{}, securityPolicies...)
+			defer g.DeleteConfigOrFail(t, ns, ingressCfgs...)
+
+			var a, b echo.Instance
+			echoboot.NewBuilderOrFail(ctx, ctx).
+				With(&a, util.EchoConfig("a", ns, false, nil, g, p)).
+				With(&b, util.EchoConfig("b", ns, false, nil, g, p)).
+				BuildOrFail(t)
+
+			// These test cases verify in-mesh traffic doesn't need tokens.
+			testCases := []authn.TestCase{
+				{
+					Name: "in-mesh-with-expired-token",
+					Request: connection.Checker{
+						From: a,
+						Options: echo.CallOptions{
+							Target:   b,
+							PortName: "http",
+							Scheme:   scheme.HTTP,
+							Headers: map[string][]string{
+								authHeaderKey: {"Bearer " + jwt.TokenExpired},
+							},
+						},
+					},
+					ExpectResponseCode: response.StatusCodeOK,
+				},
+				{
+					Name: "in-mesh-without-token",
+					Request: connection.Checker{
+						From: a,
+						Options: echo.CallOptions{
+							Target:   b,
+							PortName: "http",
+							Scheme:   scheme.HTTP,
+						},
+					},
+					ExpectResponseCode: response.StatusCodeOK,
+				},
+			}
+			for _, c := range testCases {
+				t.Run(c.Name, func(t *testing.T) {
+					retry.UntilSuccessOrFail(t, c.CheckAuthn,
+						retry.Delay(250*time.Millisecond), retry.Timeout(30*time.Second))
+				})
+			}
+
+			// These test cases verify requests go through ingress will be checked for validate token.
+			ingTestCases := []struct {
```

Comment:
authn_util.go test case and this inline one create some in-consistencies of how we organize the test.